### PR TITLE
Fee calculation break and gas limit to take gas deposit into account

### PR DESCRIPTION
**Pull request by claudiucelfilip**
- uses bignumber.js to handle the transformation of amount (limit, deposit or amount) input.
User input fields expect PERLS and the resulting request (and payload) needs to be in KENs. #62 
Previously, this went through parseFloat or parseInt and thus caused errors when trying to convert to JSBI.
- gas deposit is now passed into the 25/50/100% gas limit buttons. #61 
- small fix for resetting the input stake value switching between place and withdraw

### CODE REVIEW

File: src/components/common/core.tsx [hunk 1]
```diff
@@ -2,6 +2,7 @@ import styled from "styled-components";
 import { Flex } from "@rebass/grid";
 import Dropdown, { Option } from "react-dropdown";
 import JSBI from "jsbi";
+import BigNumber from "bignumber.js";
```
Comment (ryanswart) [inline src/components/common/core.tsx]: Hmm, not sure about this one, not great that we're using 2 bignum libraries

File: src/components/validator/StakeCard.tsx [hunk 2]
```diff
@@ -4,12 +4,12 @@ import { useState } from "react";
...
     RoundButton,
-    ErrorMessage,
     LargeInput,
-    formatBalance
+    formatBalance,
+    inputToKens
 } from "../common/core";
 import { Card, CardHeader, CardTitle, CardBody } from "../common/card";
-
+import BigNumber from "bignumber.js";
```
Comment (ryanswart) [inline src/components/validator/StakeCard.tsx]: Do we use this library in this file? I dont see it

**Review by ryanswart — DISMISSED** [review]

File: src/components/common/core.tsx (hunk 1 above)
Comment (claudiucelfilip) [reply src/components/common/core.tsx]: We need BigNumber to handle floats from user inputs. 
We might be able to remove JSBI from Lens as the new changes for wavelet-client should allow us to pass in strings.

File: src/components/validator/StakeCard.tsx (hunk 2 above)
Comment (claudiucelfilip) [reply src/components/validator/StakeCard.tsx]: No. It was omitted after a small refactor.

File: src/components/common/core.tsx (hunk 1 above)
Comment (ryanswart) [reply src/components/common/core.tsx]: Interesting, JSBI is also a larger dependency than bignumber.js. How big of a refactor do you estimate ripping out JSBI is for lens? Although I guess if it's a dep for wavelet-client, we're not going to be seeing much space saving

Comment (claudiucelfilip) [reply src/components/common/core.tsx]: Doesn't look trivial, a quick search yielded about 55 JSBI calls. It worth it to simplify Lens, but it'll take some time and has a risk of bugs.

Comment (ryanswart) [reply src/components/common/core.tsx]: Ok, sure, lets just merge this one then

**Review by ryanswart — APPROVED** [review]
LGTM, but lets leave a larger issue around Bignumber.js and JSBI somewhere

**Comment by claudiucelfilip** [comment]
> LGTM, but lets leave a larger issue around Bignumber.js and JSBI somewhere

Created #64 for that.

**Pull request merged**